ext/mysqli/mysqli.stub.php

```diff
@@ -0,0 +1,773 @@
+<?php
+
+class mysqli_driver
+{
+}
+
+class mysqli
+{
+    /**
+     * @param ?string $host
+     * @param ?string $user
+     * @param ?string $password
+     * @param ?string $database
+     * @param ?int $port
+     * @param ?string $socket
+     */
+    public function __construct(?string $host = null, ?string $user = null, ?string $password = null, ?string $database = null, ?int $port = null, ?string $socket = null);
+
+    /**
+     * @param ?bool $mode
+     *
+     * @return bool
+     */
+    public function autocommit(?bool $mode);
+
+    /**
+     * @param ?int $flags
+     * @param ?string $name
+     *
+     * @return bool
+     */
+	public function begin_transaction(?int $flags = 0, ?string $name = '');
+
+	/**
+	 * @param ?string $user
+	 * @param ?string $password
+	 * @param ?string $database
+	 *
+	 * @return bool
+	 */
+	public function change_user(?string $user, ?string $password, ?string $database);
+
+	/**
+	 * @return string
+	 */
+	public function character_set_name();
+
+    /**
+     * @return bool
+     */
+	public function close();
+
+	/**
+	 * @param ?int $flags
+	 * @param ?string $name
+	 *
+	 * @return bool
+	 */
+	public function commit(?int $flags = 0, ?string $name = '');
+
+    /**
+     * @param ?string $host
+     * @param ?string $user
+     * @param ?string $password
+     * @param ?string $database
+     * @param ?int $port
+     * @param ?string $socket
+     *
+     * @return mysqli
+     */
+	public function connect(?string $host = null, ?string $user = null, ?string $password = null, ?string $database = null, ?int $port = null, ?string $socket = null);
+
+    /**
+     * @return bool
+     */
+	public function dump_debug_info();
+
+	/**
+	 * @param ?string $debug_options
+	 * @return bool
+	 */
+	public function debug(?string $debug_options);
+
+    /**
+     * @return object
+     */
+	public function get_charset();
+
+	/**
+	 * @return string
+	 */
+	public function get_client_info();
+
+	/**
+	 * @return array
+	 */
+	public function get_connection_stats();
+
+	/**
+	 * @return string
+	 */
+	public function get_server_info();
+
+	/**
+	 * @return mysqli_warning
+	 */
+	public function get_warnings();
+
+	/**
+	 * @return mysqli
+	 */
+	public function init();
+
+	/**
+	 * @param ?int $connection_id
+	 *
+	 * @return bool
+	 */
+	public function kill(?int $connection_id);
+
+	/**
+	 * @param ?string $query
+	 *
+	 * @return bool
+	 */
+	public function multi_query(?string $query);
+
+	/**
+	 * @return bool
+	 */
+	public function more_results();
+
+	/**
+	 * @return bool
+	 */
+	public function next_result();
+
+	/**
+	 * @return bool
+	 */
+	public function ping();
+
+	/**
+	 * @param ?array &$read
+	 * @param ?array &$write
+	 * @param ?array &$error
+	 * @param ?int $sec
+	 * @param ?int $usec
+	 *
+	 * @return int
+	 */
+	public static function poll(?array &$read, ?array &$write, ?array &$error, ?int $sec, ?int $usec = 0);
```

Comment:
`$error`, `$sec` are not nullable,  and the method can also return `false`. Furthermore, I am not sure if any of these parameters are passed by reference. 🤔 Usually, when that's the case, `z` is used in the ZPP string.

Reply:
Whether an argument is passed by reference is codified in the arginfo (first argument). In this case https://github.com/php/php-src/pull/4913/files#diff-789a46e23b7b31e8bf2f4dbe51d753bfL62-L68.

Reply:
Thanks for the additional information!